keep-ui/app/settings/users-settings.tsx

```diff
@@ -39,61 +41,50 @@ export default function UsersSettings({
   const apiUrl = getApiURL();
   const { data, error, isLoading } = useSWR<User[]>(
     selectedTab === "users" ? `${apiUrl}/settings/users` : null,
-    (url) => fetcher(url, accessToken),
+    async (url) => {
+      const response = await fetcher(url, accessToken);
+      setUsers(response); // Update users state
+      return response;
+    },
     { revalidateOnFocus: false }
   );
 
+
   const { data: configData } = useSWR<Config>("/api/config", fetcher, {
     revalidateOnFocus: false,
   });
 
+  const [isAddUserModalOpen, setAddUserModalOpen] = useState(false);
```

Comment:
Rather than rebuilding custom logic to handle modals and menus, there should be a hook called `useDisclosure` to help us out. 

Here's an [OS example](https://github.com/mantinedev/mantine/blob/master/packages/@mantine/hooks/src/use-disclosure/use-disclosure.ts) from mantine.

Reply:
dam thats good